File: src/main/java/me/xdrop/nightowl/settings/NightOwlConfigurable.java
package me.xdrop.nightowl.settings;

import com.intellij.openapi.options.Configurable;
import me.xdrop.nightowl.settings.form.NightOwlSettingsForm;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class NightOwlConfigurable implements Configurable {
    private NightOwlSettingsForm settingsForm;

    @Nls
    @Override
    public String getDisplayName() {
        return "Night Owl";
    }

    @Nullable
    @Override
    public JComponent createComponent() {
        settingsForm = new NightOwlSettingsForm();
        return settingsForm.getRootPanel();
    }

    @Override
    public boolean isModified() {
        return settingsForm != null && settingsForm.isModified();
    }

    @Override
    public void reset() {
        settingsForm.reset();
    }

    @Override
    public void apply() {
        NightOwlSettings instance = NightOwlSettings.getInstance();
        instance.setEnabled(settingsForm != null && settingsForm.isNightOwlEnabled());
        instance.setOverrideAppearance(settingsForm != null && settingsForm.isAppearanceOverriden());
    }
}

File: src/main/kotlin/me/xdrop/nightowl/settings/NightOwlConfigurable.kt
package me.xdrop.nightowl.settings

import com.intellij.openapi.options.Configurable
import com.intellij.openapi.options.ConfigurationException
import me.xdrop.nightowl.settings.form.NightOwlSettingsForm
import org.jetbrains.annotations.Nls
import javax.swing.JComponent

class NightOwlConfigurable : Configurable {
    private var settingsForm: NightOwlSettingsForm? = null

    @Nls
    override fun getDisplayName() = "Night Owl"

    override fun createComponent(): JComponent? {
        settingsForm = settingsForm ?: NightOwlSettingsForm()
        return settingsForm?.component()
    }

    override fun isModified(): Boolean {
        return settingsForm?.isModified ?: false
    }

    @Throws(ConfigurationException::class)
    override fun apply() {
        val settings = NightOwlSettings.instance
        settings.isAppearanceSettingsEnabled = settingsForm?.isAppearanceSettingsEnabled ?: false
    }

    override fun reset() {
        settingsForm?.reset()
    }

    override fun disposeUIResources() {
        settingsForm = null
    }
}
